sound/soc/sof/intel/hda-loader.c

```diff
@@ -27,7 +27,7 @@ static int cl_stream_prepare(struct snd_sof_dev *sdev, unsigned int format,
 			     unsigned int size, struct snd_dma_buffer *dmab,
 			     int direction)
 {
-	struct hdac_ext_stream *dsp_stream = NULL;
+	struct hdac_ext_stream *dsp_stream;
```

Comment:
to make this even more obvious you could modify the if below to:
```
if (direction != SNDRV_PCM_STREAM_PLAYBACK) {
	dev_err(sdev->dev, "error: code loading DMA is playback only\n");
	return -EINVAL;
}
dsp_stream = hda_dsp_stream_get(sdev, direction);
```
then dsp_stream initialisation will become even more obvious.